File: src/main/java/frc/team5190/feeder/DisplayFrame.java
package frc.team5190.feeder;

import java.awt.Color;

import javax.swing.*;

public class DisplayFrame extends JFrame{
    private static final long serialVersionUID = -924699517006012122L;

    DisplayFrame() {
        this.setTitle("Remote Path Generator Status");
        this.setSize(450, 300);
        setBackground(Color.RED);
        this.setAlwaysOnTop(true);
        this.setUndecorated(true);
        this.getRootPane().setWindowDecorationStyle(JRootPane.FRAME);
        this.setOpacity(0.5f);
        this.setLocationRelativeTo(null);
        this.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        this.setVisible(true);
    }

    /**
     * Sets the background of the frame. Display color codes: <ul>
     * 		<li>RED: NT Disconnected</li>
     * 		<li>GREEN: NT Connected</li>
     * 		<li>YELLOW: Currently generating trajectories</li>
     * 		<li>BLACK: Fatal error, one or more trajectories failed to generate</li>
     */
    public void setBackground(Color color) {
        if(getBackground()!=Color.BLACK) {
            this.getContentPane().setBackground(color);
        }
    }

    public Color getBackground() {
        return this.getContentPane().getBackground();
    }

}

File: src/main/kotlin/frc/team5190/feeder/DisplayFrame.kt
package frc.team5190.feeder

import java.awt.Color

import javax.swing.*

class DisplayFrame internal constructor() : JFrame() {

    init {
        this.title = "Remote Path Generator Status"
        this.setSize(450, 300)
        background = Color.RED
        this.isAlwaysOnTop = true
        this.isUndecorated = true
        this.getRootPane().windowDecorationStyle = JRootPane.FRAME
        this.opacity = 0.5f
        this.setLocationRelativeTo(null)
        this.defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
        this.isVisible = true
    }

    /**
     * Sets the background of the frame. Display color codes:
     *  * RED: NT Disconnected
     *  * GREEN: NT Connected
     *  * YELLOW: Currently generating trajectories
     *  * BLACK: Fatal error, one or more trajectories failed to generate
     */
    override fun setBackground(color: Color) {
        if (background !== Color.BLACK) {
            this.contentPane.background = color
        }
    }

    override fun getBackground(): Color {
        return this.contentPane.background
    }

    companion object {
        private val serialVersionUID = -924699517006012122L
    }

}